src/components/dashboard/dashboard.spec.ts

```diff
@@ -121,4 +123,128 @@ describe('Dashboard', () => {
 
 function getPixels(cssValue: string): number {
     return parseInt(cssValue.replace('px', ''));
-}
\ No newline at end of file
+}
+
+@Component({
+    selector: 'app-ux-dashboard-layout',
+    template: `<ux-dashboard
+                    [options]="options"
+                    [(layout)]="layout"
+                    (layoutChange)="onLayoutChange()">
+
+                    <ux-dashboard-widget
+                        id="run-widget"
+                        name="Runs">
+                    </ux-dashboard-widget>
+
+                    <ux-dashboard-widget
+                        id="purpose-widget"
+                        name="Purpose">
+                    </ux-dashboard-widget>
+
+                    <ux-dashboard-widget
+                        id="host-widget"
+                        name="Host">
+                    </ux-dashboard-widget>
+
+            </ux-dashboard>
+`
+})
+export class DashboardWithInitialLayoutTestComponent {
+
+    options: DashboardOptions = {
+        columns: 3,
+        padding: 10,
+        rowHeight: 300,
+        emptyRow: false,
+        minWidth: 187
+    };
+
+    layout: ReadonlyArray<DashboardLayoutData> = [
+        { id: 'run-widget', col: 0, row: 0, colSpan: 2, rowSpan: 1},
+        { id: 'purpose-widget', col: 0, row: 2, colSpan: 1, rowSpan: 1},
+        { id: 'host-widget', col: 0, row: 3, colSpan: 3, rowSpan: 1},
+    ];
+
+    @ViewChildren(DashboardWidgetComponent) widgets: QueryList<DashboardWidgetComponent>;
+
+    onLayoutChange(): void {
+
+    }
+
+}
+
+describe('Dashboard with initial layout', () => {
+    let component: DashboardWithInitialLayoutTestComponent;
+    let fixture: ComponentFixture<DashboardWithInitialLayoutTestComponent>;
+    let nativeElement: HTMLElement;
+    let layoutChangeSpy: jasmine.Spy;
+
+    beforeEach(async(() => {
+        TestBed.configureTestingModule({
+            imports: [DashboardModule],
+            declarations: [DashboardWithInitialLayoutTestComponent]
+        })
+            .compileComponents();
+    }));
+
+    beforeEach(async () => {
+        fixture = TestBed.createComponent(DashboardWithInitialLayoutTestComponent);
+        component = fixture.componentInstance;
+        nativeElement = fixture.nativeElement;
+        layoutChangeSpy = spyOn(component, 'onLayoutChange');
+
+        fixture.detectChanges();
+        await fixture.whenStable();
+    });
+
+    it('should update the widgets correctly when an initial layout value is given', async () => {
+        // check that the underlying widgets receive the correct position and size values
+
+        expect(layoutChangeSpy).not.toHaveBeenCalled();
+
+        expect(component.widgets.toArray()[0].getColumn()).toBe(0);
```

Comment:
As above, could be good to add descriptions here like `.toBe(0, 'widget 0 column')`.